app/controllers/Application.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2017 Lightbend
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import javax.inject.Inject
+import play.api.cache._
+import play.api.mvc.Action
+import play.api.mvc.Controller
+import helpers.Auth0Config
+import java.security.SecureRandom
+import java.math.BigInteger
+
+class Application @Inject() (cache: CacheApi) extends Controller {
+
+  def index = Action {
+    Ok(views.html.index())
+  }
+
+  def login = Action {
+    val config = Auth0Config.get()
+    // Generate random state parameter
+    object RandomUtil {
```

Comment:
You could use an existing utility like `UUID.randomUUID` to avoid having to write a custom helper.